### PR TITLE
RDoc-1978_StudioPatchView - added Studio Patch View documentation

### PR DESCRIPTION
**Pull request by reebhub**
added studio\database\documents\patch-view

### CODE REVIEW

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+---
+
+{PANEL: Patch View}
+
+![Figure 1. Patch View](images/patch-view-1.png "Figure 1. Patch View")
+
+1. **Patch View**  
+   Click to open the patch view.  
+2. **Database**  
+   Click to choose the database you want to patch.
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:27]: =>
Select a database

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+![Figure 1. Patch View](images/patch-view-1.png "Figure 1. Patch View")
+
+1. **Patch View**  
+   Click to open the patch view.  
+2. **Database**  
+   Click to choose the database you want to patch.  
+3. **Syntax Samples**  
+   Click to display patching [syntax samples](../../../client-api/operations/patching/set-based#examples).  
+4. **Patch Box**  
+   Enter your patch in the patch box.
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:31]: => 
Enter your patch script in the ...

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+6. **Test Patch**  
+   Click to test your patch without actually applying it to your documents (see [Test Patch](../../../studio/database/documents/patch-view#test-patch) below).  
+{PANEL/}
+
+{PANEL: Patch Configuration}
+
+![Figure 2. Patch Configuration](images/patch-view-2.png "Figure 2. Patch Configuration")
+
+1. **Save Patch**  
+   Save the patch currently displayed in the patch box under a name of your choice.
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:43]: => Click to save the currently displayed patch in the local browser storage.

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+{PANEL/}
+
+{PANEL: Patch Configuration}
+
+![Figure 2. Patch Configuration](images/patch-view-2.png "Figure 2. Patch Configuration")
+
+1. **Save Patch**  
+   Save the patch currently displayed in the patch box under a name of your choice.  
+   ![Save Patch](images/patch-view-save-patch.png "Save Patch")  
+   Enter a name in the text box, and click the **Save** button again to store the patch.
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:45]: => Enter a name and click the **Save** button again to save the patch.

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+{PANEL: Patch Configuration}
+
+![Figure 2. Patch Configuration](images/patch-view-2.png "Figure 2. Patch Configuration")
+
+1. **Save Patch**  
+   Save the patch currently displayed in the patch box under a name of your choice.  
+   ![Save Patch](images/patch-view-save-patch.png "Save Patch")  
+   Enter a name in the text box, and click the **Save** button again to store the patch.  
+2. **Load Patch**  
+   Load a stored patch.
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:47]: =>
Click to load a stored patch.
Recent (unsaved) patches will also be shown/listed here.

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+1. **Save Patch**  
+   Save the patch currently displayed in the patch box under a name of your choice.  
+   ![Save Patch](images/patch-view-save-patch.png "Save Patch")  
+   Enter a name in the text box, and click the **Save** button again to store the patch.  
+2. **Load Patch**  
+   Load a stored patch.  
+   ![Load Patch](images/patch-view-load-patch.png "Load Patch")  
+   Hover over a patch name to display its preview.  
+   Click a patch name or the preview **Load** button to load the patch.
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:50]: Click a patch name.. => Click the patch name..

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+   ![Load Patch](images/patch-view-load-patch.png "Load Patch")  
+   Hover over a patch name to display its preview.  
+   Click a patch name or the preview **Load** button to load the patch.  
+3. **Patch Settings**  
+   ![Patch Settings](images/patch-view-settings.png "Patch Settings")  
+    * a. **Patch Immediately**  
+         When this option is selected, clicking the **Apply Patch** button applies 
+         the patch immediately.  
+    * b. **Define Timeout**  
+         When this option is selected, Clicking the **Apply Patch** button
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:57]: Clicking => clicking

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+   ![Patch Settings](images/patch-view-settings.png "Patch Settings")  
+    * a. **Patch Immediately**  
+         When this option is selected, clicking the **Apply Patch** button applies 
+         the patch immediately.  
+    * b. **Define Timeout**  
+         When this option is selected, Clicking the **Apply Patch** button 
+         applies the patch only after the index used by the patch has become 
+         non-stale.  
+         If the index remains stale for the duration of the set timeout, 
+         the operation is reverted and an exception is thrown.
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:61]: Operation is reverted ?
Maybe instead - the operation is not executed ?

The operation does Not start in this case at all - right ?

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+    * b. **Define Timeout**  
+         When this option is selected, Clicking the **Apply Patch** button 
+         applies the patch only after the index used by the patch has become 
+         non-stale.  
+         If the index remains stale for the duration of the set timeout, 
+         the operation is reverted and an exception is thrown.  
+         ![Patch Timeout](images/patch-view-timeout.png "Patch Timeout")  
+    * c. **Limit Number of Operations**  
+         Toggle between limiting the max number of patch operations per second, 
+         and allowing an unlimited number of patch operations per second.
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:65]: Remove this part

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+         the patch immediately.  
+    * b. **Define Timeout**  
+         When this option is selected, Clicking the **Apply Patch** button 
+         applies the patch only after the index used by the patch has become 
+         non-stale.  
+         If the index remains stale for the duration of the set timeout, 
+         the operation is reverted and an exception is thrown.  
+         ![Patch Timeout](images/patch-view-timeout.png "Patch Timeout")  
+    * c. **Limit Number of Operations**  
+         Toggle between limiting the max number of patch operations per second,
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:64]: => 
Toggle on this option to set the max number of patch operations per second.

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+         When this option is selected, Clicking the **Apply Patch** button 
+         applies the patch only after the index used by the patch has become 
+         non-stale.  
+         If the index remains stale for the duration of the set timeout, 
+         the operation is reverted and an exception is thrown.  
+         ![Patch Timeout](images/patch-view-timeout.png "Patch Timeout")  
+    * c. **Limit Number of Operations**  
+         Toggle between limiting the max number of patch operations per second, 
+         and allowing an unlimited number of patch operations per second.  
+         ![Patch Operations Limit](images/patch-view-operations-limit.png "Patch Operations Limit")
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:66]: this image caption is not visible ?
pls check all other images in the article...

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+    * a. **Patch Immediately**  
+         When this option is selected, clicking the **Apply Patch** button applies 
+         the patch immediately.  
+    * b. **Define Timeout**  
+         When this option is selected, Clicking the **Apply Patch** button 
+         applies the patch only after the index used by the patch has become 
+         non-stale.  
+         If the index remains stale for the duration of the set timeout, 
+         the operation is reverted and an exception is thrown.  
+         ![Patch Timeout](images/patch-view-timeout.png "Patch Timeout")
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:62]: image caption not visible ?

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+
+{PANEL: Apply Patch}
+
+![Run Patch](images/patch-view-apply-patch.png "Run Patch")  
+
+While the patch is executed, a runtime dialog presents -  
+
+1. **Time Elapsed**  
+   Time since the operation began  
+2. **Index**
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:78]: =>
Index or Collection

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 14]
```diff
@@ -0,0 +1,128 @@
...
+{PANEL: Apply Patch}
+
+![Run Patch](images/patch-view-apply-patch.png "Run Patch")  
+
+While the patch is executed, a runtime dialog presents -  
+
+1. **Time Elapsed**  
+   Time since the operation began  
+2. **Index**  
+   Index used by the patch
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:79]: =>
The index or the collection used by the patch

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+2. **Index**  
+   Index used by the patch  
+3. **Processed**  
+   Number of documents already processed / Number of documents yet to be processed  
+4. **Processing Speed**  
+   Number of documents processed per sec  
+5. **Estimated time left**  
+6. **Query**  
+   Patch's query and update operation  
+7. **Procession Bar**
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:87]: => **Progress Bar**

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+3. **Processed**  
+   Number of documents already processed / Number of documents yet to be processed  
+4. **Processing Speed**  
+   Number of documents processed per sec  
+5. **Estimated time left**  
+6. **Query**  
+   Patch's query and update operation  
+7. **Procession Bar**  
+8. **Close**  
+   Click to **close this view** (the operation will proceed)
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:89]: Separate the lines:
=>
Click to **close this dialog**.
The operation will proceed in the background.
The dialog can be re-opened from the Studio's Notification Center.

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 17]
```diff
@@ -0,0 +1,128 @@
...
+4. **Processing Speed**  
+   Number of documents processed per sec  
+5. **Estimated time left**  
+6. **Query**  
+   Patch's query and update operation  
+7. **Procession Bar**  
+8. **Close**  
+   Click to **close this view** (the operation will proceed)  
+9. **Abort**  
+   Click to **abort the operation** (the operation will be reverted)
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:91]: Separate the lines:
=>
Click to **abort the patch operation**.
The operation will be reverted.

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 18]
```diff
@@ -0,0 +1,128 @@
...
+7. **Procession Bar**  
+8. **Close**  
+   Click to **close this view** (the operation will proceed)  
+9. **Abort**  
+   Click to **abort the operation** (the operation will be reverted)  
+
+{PANEL/}
+
+{PANEL: Test Patch}
+
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:96]: Missing the very first step of clicking the 'test' button in order to enter the 'test mode'....

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 19]
```diff
@@ -0,0 +1,128 @@
...
+![Test Patch](images/patch-view-test-patch.png "Test Patch")  
+
+1. **Document**  
+    Enter the ID of a document that the patch is meant to update, and click **Load Document**.  
+    The test will patch a copy of the document, leaving the original document unharmed.  
+2. **Test**  
+   Click to apply the patch to the document.  
+3. **Done**  
+   Click to close the test view and return to the patch view.
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:105]: This 'Done' part should come after the 'After' part..

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 20]
```diff
@@ -0,0 +1,128 @@
...
+
+{PANEL/}
+
+{PANEL: Test Patch}
+
+![Test Patch](images/patch-view-test-patch.png "Test Patch")  
+
+1. **Document**  
+    Enter the ID of a document that the patch is meant to update, and click **Load Document**.  
+    The test will patch a copy of the document, leaving the original document unharmed.
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:101]: remove this sentence from here

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 21]
```diff
@@ -0,0 +1,128 @@
...
+   Click to **abort the operation** (the operation will be reverted)  
+
+{PANEL/}
+
+{PANEL: Test Patch}
+
+![Test Patch](images/patch-view-test-patch.png "Test Patch")  
+
+1. **Document**  
+    Enter the ID of a document that the patch is meant to update, and click **Load Document**.
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:100]: =>
Enter a document ID on which to run the patch in the test mode and click Load Document.
The document preview will show in the 'Before' area on the left side.

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 22]
```diff
@@ -0,0 +1,128 @@
...
+
+{PANEL: Test Patch}
+
+![Test Patch](images/patch-view-test-patch.png "Test Patch")  
+
+1. **Document**  
+    Enter the ID of a document that the patch is meant to update, and click **Load Document**.  
+    The test will patch a copy of the document, leaving the original document unharmed.  
+2. **Test**  
+   Click to apply the patch to the document.
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:103]: =>
Click 'Test' to apply the patch to the selected document.
A copy of the document will be patched, leaving the original document **untouched**.
A preview of the document after the patch will show in the 'After' area on the right.

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 23]
```diff
@@ -0,0 +1,128 @@
...
+2. **Test**  
+   Click to apply the patch to the document.  
+3. **Done**  
+   Click to close the test view and return to the patch view.  
+4. **Before**  
+   Browse the document before the patch has been applied to it.  
+   Click Shift+F11 to expand or collapse an enlarged view of the document.  
+5. **After**  
+   Browse the document after the patch has been applied to it.  
+   See line 198 in the "after" view depicted above for an example of a text the patch updated.
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:111]: after => After

=>
See line 198 in the "After" view depicted above for an example of a text modified by the patch.

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown (hunk 19 above)
Comment (Danielle9897) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:105]: =>
Click 'Done' to close the test view/mode and return to the patch view

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 24]
```diff
@@ -0,0 +1,139 @@
...
+
+1. **Document**  
+   Enter a document ID on which to run the patch in the test mode and click Load Document.  
+   The document preview will show in the 'Before' area on the left side.  
+2. **Test**  
+   Click 'Test' to apply the patch to the selected document.  
+   A copy of the document will be patched, leaving the original document unchanged.  
+   A preview of the document after the patch will show in the 'After' area on the right.  
+3. **Done**  
+   Click 'Done' to close the test view and return to the patch view.
```
Comment (Danielle9897) [inline Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:116]: This 'Done' part should come after the 'After' part

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown [hunk 25]
```diff
@@ -0,0 +1,139 @@
...
+   ![Patch Settings](images/patch-view-settings.png "Patch Settings")  
+    * a. **Patch Immediately**  
+         When this option is selected, clicking the **Apply Patch** button applies 
+         the patch immediately.  
+    * b. **Define Timeout**  
+         When this option is selected, clicking the **Apply Patch** button 
+         applies the patch only after the index used by the patch has become 
+         non-stale.  
+         If the index remains stale for the duration of the set timeout, 
+         the operation is canceled and an exception is thrown.
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown:62]: you mean "canceled"? seems like Grammarly, on behalf of all Americans, prefers it this way.

File: Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown (hunk 24 above)
Comment (reebhub) [reply Documentation/5.2/Raven.Documentation.Pages/studio/database/documents/patch-view.markdown]: done

**Review by Danielle9897 — APPROVED** [review]

**Pull request merged**